dbms/src/IO/ReadHelpers.h

```diff
@@ -955,28 +955,78 @@ inline T parse(const char * data, size_t size)
     return res;
 }
 
+template <typename T>
+inline std::enable_if_t<!is_integral_v<T>, void>
+readTextWithSuffix(T & x, ReadBuffer & buf) { readText(x, buf); }
+
+template <typename T>
+inline std::enable_if_t<is_integral_v<T>, void>
+readTextWithSuffix(T & x, ReadBuffer & buf)
+{
+    readIntText(x, buf);
+    if (buf.eof())
+        return;
+
+    auto F = [&buf, &x] (long long base, int power_of_two) mutable
```

Comment:
Not `long long` because in fact we use `unsigned long long`.
But better to use `UInt64`.